cmd/bclient/main.go

```diff
@@ -166,8 +172,107 @@ func doUpload(item string, files string) {
 	// TODO: cleanup uploads on success
 }
 
-func doGet(item string, files string) {
-	fmt.Printf("Item = %s\n", item)
+//  doGet , given only an item, returns all the files in that item.
+//  Given one or more files in the item, it returns only them
+
+func doGet(item string, files []string) {
+	var json *jason.Object
+	var jsonFetchErr error
+	filesToGet := make(chan string)
+	var getFileDone sync.WaitGroup
+
+	// if file or dir exists in target path named after the item, give error mesg and exit
+	pathPrefix := path.Join(*fileroot, item)
+
+	_, err := os.Stat(pathPrefix)
+
+	if err == nil {
+		// file already exists
+		fmt.Printf("Error: target %s already exists", pathPrefix)
+		return
+	}
+
+	// set up communication to the bendo server, and init local and remote filelists
+
+	thisItem := bserver.New(*server, item, *fileroot)
+	fileLists := fileutil.NewLists(*fileroot)
+
+	// Fetch Item Info from bserver
+	json, jsonFetchErr = thisItem.GetItemInfo()
+
+	// If not found or error, we're done
+
+	switch {
+	case jsonFetchErr == bserver.ErrNotFound:
+		fmt.Printf("\n Item %s was not found on server %\n", item, *server)
+		return
+	case jsonFetchErr != nil:
+		fmt.Println(jsonFetchErr)
+		return
+	}
+
+	// if item only, get all of the files; otherwise, only those asked for
+
+	if len(files) == 0 {
+		fileLists.BuildLocalList(json)
+	} else {
+		fileLists.BuildRemoteList(json)
+		fileLists.BuildLocalFromFiles(files)
+	}
+
+	// At this point, the local list contains files, verified to exist on server
+
+	// set up our barrier, that will wait for all the file chunks to be uploaded
```

Comment:
some of these comments refer to uploading. Isn't this function downloading content?